codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -731,6 +731,12 @@ protected Http2FrameWriter frameWriter() {
      */
     private ChannelFuture resetUnknownStream(final ChannelHandlerContext ctx, int streamId, long errorCode,
                                              ChannelPromise promise) {
+
+        if (!connection().streamMayHaveExisted(streamId)) {
```

Comment:
is this necessary after the [recent change to streamError](https://github.com/netty/netty/commit/c321e8ea4a1349125510bd9156a4e9a4c3e32741#diff-ce33d9374618cacba50c2b6020df1c86R704), and with your changes to `Http2MultiplexCodec` in this PR? IIUC from the perspective of this class the only "invalid" way this can be called is if the "user" explicitly writes a RST_STREAM frame for a bogus stream.

Reply:
That's right, the problem is that `Http2MultiplexCodec` is doing the bad thing of explicitly writing a RST_STREAM. I added this because it was easier to test and looked like it would fix a problem in both the multiplex and original API's, but only one should be sufficient. Should we fix both places, or just `Http2MultiplexCodec`?

Reply:
I vote for just `Http2MultiplexCodec`. Better to fix the "root" cause where this feels more like a band-aid. If protecting applications from sending bogus RST is also useful in the future we can address it with a followup PR.

Reply:
done.